### PR TITLE
Require a scheduler to submit, even with --pretend enabled.

### PR DESCRIPTION
**Pull request by bdice**
## Description
This PR forces all "pretend" submissions to go through a scheduler. Environments without a scheduler such as `StandardEnvironment` are no longer able to submit in pretend mode. It should be possible to achieve similar output using the `run --pretend` functionality (environments without schedulers can still use `run`, just not `submit`).

For testing purposes, the `TestEnvironment` uses `FakeScheduler` which always just prints the submission script. Use the `TestEnvironment` with the environment variable `SIGNAC_FLOW_ENVIRONMENT="TestEnvironment"`.

## Motivation and Context
Currently users can call `python project.py submit --pretend` even when using the `StandardEnvironment`, which renders the base template with the output script. However, this isn't very useful and the current code circumvents the use of the schedulers' support for `pretend=True`, which can provide additional information such as the submission command that would be used without `--pretend`.

## Types of Changes
- [ ] Documentation update
- [ ] Bug fix
- [x] New feature
- [x] Breaking change<sup>1</sup>

<sup>1</sup>The change breaks (or has the potential to break) existing functionality.

## Checklist:
- [x] I am familiar with the [**Contributing Guidelines**](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md).
- [x] I agree with the terms of the [**Contributor Agreement**](https://github.com/glotzerlab/signac-flow/blob/master/ContributorAgreement.md).
- [x] My name is on the [list of contributors](https://github.com/glotzerlab/signac-flow/blob/master/contributors.yaml).
- [x] My code follows the [code style guideline](https://github.com/glotzerlab/signac-flow/blob/master/CONTRIBUTING.md#code-style) of this project.
- [ ] The changes introduced by this pull request are covered by existing or newly introduced tests.

If necessary:
- [ ] I have updated the API documentation as part of the package doc-strings.
- [ ] I have created a separate pull request to update the [framework documentation](https://docs.signac.io/) on [signac-docs](https://github.com/glotzerlab/signac-docs) and linked it here.
- [x] I have updated the [changelog](https://github.com/glotzerlab/signac-flow/blob/master/changelog.txt).

### CODE REVIEW

File: changelog.txt [hunk 1]
```diff
@@ -23,6 +23,7 @@ Changed
 
 - Jinja templates are indented for easier reading (#461, #495).
 - ``flow.directives`` is deprecated in favor of ``flow.FlowProject.operation.with_directives`` (#309, #502).
+- All environments require a scheduler in order to submit, even in pretend mode (#533).
```
Comment (bdice) [inline changelog.txt:26]:
```suggestion
- All environments require a scheduler in order to submit, even in pretend mode (#533).
- Submitting in pretend mode will show additional scheduler command information (#533).
```

**Comment by csadorf** [comment]
When I first read the title I thought you meant that a scheduler *must* submit even with `--pretend` enabled. The comma makes all the difference there. 😄

**Review by atravitz — APPROVED** [review]
after our conversation in dev meeting on 6/7, I think this is a good change

**Comment by bdice** [comment]
@tcmoore3 @SchoeniPhlippsn Can one of you please review this PR when you have time? Thanks!

**Review by tcmoore3 — APPROVED** [review]
LGTM. This is certainly an improvement in the logic w.r.t. how `pretend` is used. 

For reference, I talked to Bradley offline about actually removing the `pretend` argument to `_submit_operations()`, which was my only concern.

**Comment by tcmoore3** [comment]
> When I first read the title I thought you meant that a scheduler _must_ submit even with `--pretend` enabled. The comma makes all the difference there. 😄

Same!

**Pull request merged**